scanpy/plotting/_baseplot_class.py

```diff
@@ -109,6 +111,11 @@ def __init__(
             layer=layer,
             gene_symbols=gene_symbols,
         )
+        if len(self.categories) > self.MAX_NUM_CATEGORIES:
+            raise ValueError(
+                f"Over {self.MAX_NUM_CATEGORIES} categories found. "
+                f"Plot would be too large."
+            )
```

Comment:
I'm not sure we should restrict this. If someone wants a huge plot, why should we stop them?

Reply:
I changed this to a warning in case someone ends up making a plot with lot of categories by accident.